examples/noname_full_flow.rs

```diff
@@ -0,0 +1,154 @@
+#![allow(non_snake_case)]
+#![allow(non_camel_case_types)]
+#![allow(clippy::upper_case_acronyms)]
+///
+/// This example performs the full flow:
+/// - define the circuit to be folded
+/// - fold the circuit with Nova+CycleFold's IVC
+/// - generate a DeciderEthCircuit final proof
+/// - generate the Solidity contract that verifies the proof
+/// - verify the proof in the EVM
+///
+use ark_bn254::{constraints::GVar, Bn254, Fr, G1Projective as G1};
+use noname::backends::r1cs::R1csBn254Field;
+
+use ark_groth16::Groth16;
+use ark_grumpkin::{constraints::GVar as GVar2, Projective as G2};
+
+use folding_schemes::{
+    commitment::{kzg::KZG, pedersen::Pedersen},
+    folding::nova::{
+        decider_eth::{prepare_calldata, Decider as DeciderEth},
+        Nova, PreprocessorParam,
+    },
+    frontend::{
+        noname::{utils::NONAME_CIRCUIT_EXTERNAL_INPUTS, NoNameFCircuit},
+        FCircuit,
+    },
+    transcript::poseidon::poseidon_canonical_config,
+    Decider, FoldingScheme,
+};
+use std::time::Instant;
+
+use solidity_verifiers::{
+    evm::{compile_solidity, Evm},
+    utils::get_function_selector_for_nova_cyclefold_verifier,
+    verifiers::nova_cyclefold::get_decider_template_for_cyclefold_decider,
+    NovaCycleFoldVerifierKey,
+};
+
+fn main() {
+    // set the initial state
+    let z_0 = vec![Fr::from(2), Fr::from(5)];
+
+    // set the external inputs to be used at each step of the IVC, it has length of 10 since this
+    // is the number of steps that we will do
+    let external_inputs = vec![
+        vec![Fr::from(8u32), Fr::from(2u32)],
+        vec![Fr::from(40), Fr::from(5)],
+    ];
+
+    // initialize the noname circuit
+    let f_circuit_params = (NONAME_CIRCUIT_EXTERNAL_INPUTS.to_owned(), 2, 2);
```

Comment:
I was confused by the upper-case constant here thinking that was some constant of the logic, but after diving in I found out that this constant is in fact an example of a test circuit defined with Noname's language.
Since it only takes 4 lines, how would you see to paste here the circuit so the dev who reads the example sees at first sight that this is the circuit code?
Also in this way the test circuits of Noname don't have to be exposed as global constants in the frontend/noname package.

Reply:
Yup, did it [here](https://github.com/privacy-scaling-explorations/sonobe/blob/d2db27d434dd70c5078cbd1738a9e8e6e0577c4d/examples/noname_full_flow.rs#L38)